enhancements/20250203-third-party-storage-support.md

```diff
@@ -0,0 +1,225 @@
+  harvester-system                  cdi-operator-7b6695c56d-l5lx4                            1/1     Running     0             17h
+  harvester-system                  cdi-uploadproxy-68d59ccffb-qdvns                         1/1     Running     0             17h
+  ```
+
+- Ensure the fleet did not complain anythin with the new CDI. (Both fresh install and upgrade)
+  You can check the fleet status by running the following command:
+  ```shell
+  $ kubectl get bundle -A |grep harvester
+  fleet-local   mcc-harvester                                 1/1
+  fleet-local   mcc-harvester-crd                             1/1
+  ```
+
+#### Support Matrix
+
+| Storage Solution | VM Image | VM Rook Disk | VM Data Disk | Volume Export To VM Image | VM Template Generator | VM Snapshot | VM Backup |
+| --- | --- | --- | --- | --- | --- | --- | --- |
+| Longhorn v2 Data Engine | &#10004; | &#10004; | &#10004; | &#10004; | &#10004; | &#10006; | &#10006; |
+| Rook (RBD) | &#10004; | &#10004; | &#10004; | &#10004; | &#10004; | &#10004; | &#10006; |
+| LVM | &#10004; | &#10004; | &#10004; | &#10004; | &#10004; | &#10004; | &#10006; |
+| NFS | &#10004; | &#10004; | &#10004; | &#10004; | &#10004; | &#10006; | &#10006; |
+
+**NOTE:** We only need to test the RBD of Rook, not the CephFS.
+
+The general test items are the following. We should test the same items for the different storage solutions mentioned above.
+
+- VM Image
+  - Ensure the VM image can be created through download.
+  - Ensure the VM image can be created through upload.
+  - Ensure the user can download the VM image locally. (means users' local machine/laptop)
+
+- VM Rook Disk
+  - Ensure we could boot from the RAW/QCOW2 VM Image.
+  - Ensure the VM can be created with the above VM image (which means the root disk is created with the 3rd-party storage solution)
+
+- VM Data Disk
+  - Ensure the Data Disk can be created with the VM with the host disk path. (Means configure before VM start)
+  - Ensure the Data Disk can be hotplugged to the VM. (Means configure after VM start)
+
+- Volume Export To VM Image
+  - Ensure the volume can be exported to VM Image with any 3rd-party storage solution.
+
+**NOTE**: The volume should be exported to any 3rd-party storage solution but only the longhorn v1 volume can be exported to longhorn v1 VM Image.
+
+- VM Template Generator
+  - Ensure the VM can generate the VM template with various disk configurations. (Like a mix of the Longhorn v1 and LVM)
+  - Ensure the above VM template can be used to create the VM.
+
+- VM snapshot
+  - Ensure the VM can take the snapshot with various disk configurations. (Like a mix of the Longhorn v1 and LVM)
+  - Ensure the above VM snapshot can be restored to the existing VM or new VM.
+
+- VM Backup
+  - We did not support the VM backup with any third-party storage solution. So, ensure this action item does not appear in the UI.
+
```

Comment:
Some suggestions regarding the test plan
- SC creation.
- When an Image is created/deleted, corresponding volumes in 3rd party storage should be created/deleted. 
- VM live migration should work for RWX volumes.
- Harvester upgrade should work when there are VMs booted with 3rd party storage.

Reply:
updated, thanks for the suggestion